### PR TITLE
fix build error

### PR DESCRIPTION
**Pull request by deepaksrma**
- this commit fix the build error of commit 057dec00ab9911dbcbf43379e11fc923ab31d056

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>

### CODE REVIEW

File: os/drivers/kernel_test/Make.defs [hunk 1]
```diff
@@ -19,7 +19,11 @@
 include kernel_test$(DELIM)kernel$(DELIM)Make.defs
 include kernel_test$(DELIM)network$(DELIM)Make.defs
 
+ifeq ($(CONFIG_DRIVERS_KERNEL_TEST),y)
 CSRCS += kernel_test_drv.c
+else ifeq ($(CONFIG_TC_NET_PBUF),y)
+CSRCS += kernel_test_drv.c
```
Comment (pillip8282) [inline os/drivers/kernel_test/Make.defs:25]: CONFIG_DRIVERS_KERNEL_TEST and CONFIG_TC_NET_PBUF are different layer. so this code seems not good to me. You'd better to add dependency between both config in Kconfig. Please check 'depends on' and 'select' command in Kconfig

**Review by pillip8282 — CHANGES_REQUESTED** [review]

File: os/drivers/kernel_test/Make.defs (hunk 1 above)
Comment (deepaksrma) [reply os/drivers/kernel_test/Make.defs]: changed. check now

**Review by thapav — COMMENTED** [review]
LGTM

**Comment by Taejun-Kwon** [comment]
Problem is happened because pbuf using kernel_test_drv which added for kernel test.
So below is needed
1. rename kernel_test_drv to something common name for both of kernel & network. Of course another module can use it.
    Or network should have its own virtual driver for test
2.  If we allow kernel_test_drv for all other module, CONFIG_DRIVERS_KERNEL_TEST also need to be renamed.
3. Each of module should enable that config in their defconfig.

**Review by Taejun-Kwon — CHANGES_REQUESTED** [review]
please see my comment.

File: apps/examples/testcase/le_tc/network/Kconfig [hunk 2]
```diff
@@ -53,6 +53,7 @@ config TC_NET_SOCKET
 config TC_NET_PBUF
 	bool "pbuf() api "
 	default n
+	select DRIVERS_KERNEL_TEST
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/Kconfig:56]: Please split the commit for this.

File: os/drivers/kernel_test/Make.defs [hunk 3]
```diff
@@ -19,7 +19,9 @@
 include kernel_test$(DELIM)kernel$(DELIM)Make.defs
 include kernel_test$(DELIM)network$(DELIM)Make.defs
 
+ifeq ($(CONFIG_DRIVERS_KERNEL_TEST),y)
```
Comment (sunghan-chang) [inline os/drivers/kernel_test/Make.defs:22]: Please cover above  as well.
```
ifeq (...)
include
include
CSRCS +=
DEPPATH +=
VPATH +=
endif
```

**Review by sunghan-chang — COMMENTED** [review]
And please cleanup including header in kernel_test_drv.c

**Comment by sunghan-chang** [comment]
@Taejun-Kwon @deepaksrma I agree your opinion. Let's rename it but let's do that with another PR. Because it's not for resolving build break. Resolving build break is more important so that let's finish another thing first and merge this first.

**Comment by sunghan-chang** [comment]
@deepaksrma Let's cleanup including header with another pr.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**